examples/utils/pkg/utils/buffer_to_obj.go

```diff
@@ -0,0 +1,27 @@
+// Package utils provides the general utilities functionalities for the integrations
+package utils
+
+import (
+	"bytes"
+	"encoding/json"
+	"strings"
+)
+
+// UnmarshalBufferToStruct unmarshals a buffer to a slice of structs with type T
+func UnmarshalBufferToStruct[T any](buf bytes.Buffer) ([]T, error) {
+	var outputSlice []T
+	strBuf := buf.String()
+	sliceStrBuf := strings.Split(strBuf, "\n")
+	for _, item := range sliceStrBuf {
+		if item == "" {
+			continue
+		}
+		var m T
```

Comment:
In general I would avoid to use vars with names such as "m" as they are rather obscure and can be confusing.
That's just a personal opinion, feel free to keep using them if you think that they are a better fit in this case.